service/worker/sysworkflow/constants.go

```diff
@@ -24,62 +24,45 @@ import (
-	SystemWorkflowIDTag = "system-workflow-id"
-	// HandledSignalCount counter of number of signals processed by system workflow
-	HandledSignalCount = "handled-signal"
-	// UnknownSignalTypeErr counter of number of unknown signals received by system workflow
-	UnknownSignalTypeErr = "unknown-signal-err"
-	// ArchivalFailureErr counter of number of archival activity failures
-	ArchivalFailureErr = "archival-failure"
-	// BackfillFailureErr counter of number of backfill activity failures
-	BackfillFailureErr = "backfill-failure"
-	// ChannelClosedUnexpectedlyError counter of number of unexpected channel closes in system workflow
-	ChannelClosedUnexpectedlyError = "channel-closed-unexpectedly-err"
-	// ArchivalActivityFnName name of archival activity function
-	ArchivalActivityFnName = "ArchivalActivity"
-	// BackfillActivityFnName name of backfill activity function
-	BackfillActivityFnName = "BackfillActivity"
-	// SystemWorkflowFnName name of system workflow function
-	SystemWorkflowFnName = "SystemWorkflow"
-	// WorkflowStartToCloseTimeout is the time for the workflow to finish
-	WorkflowStartToCloseTimeout = time.Hour * 24 * 30
-	// DecisionTaskStartToCloseTimeout is the time for decision to finish
-	DecisionTaskStartToCloseTimeout = time.Minute
-	// DomainIDTag tag which identifies the domainID of an archived history
-	DomainIDTag = "domainID"
-	// WorkflowIDTag tag which identifies the workflowID of an archived history
-	WorkflowIDTag = "workflowID"
-	//RunIDTag tag which identifies the runID of an archived history
-	RunIDTag = "runID"
-	// BucketNameTag tag which identifies the bucket name
-	BucketNameTag = "bucket-name"
-)
-
-// RequestType is the type for signals that can be sent to system workflows
-type RequestType int
+type requestType int
 
 const (
-	archivalRequest RequestType = iota
+	archivalRequest requestType = iota
 	backfillRequest
 )
 
 type contextKey int
 
 const (
-	blobstoreClientKey contextKey = iota
-	frontendClientKey
+	sysWorkerContainerKey contextKey = iota
+)
+
+const (
+	// SystemDomainName is domain name for all system workflows
+	SystemDomainName = "cadence-system"
+
+	workflowIDPrefix                    = "cadsys-wf"
+	decisionTaskList                    = "cadsys-decision-tl"
+	workflowStartToCloseTimeout         = time.Hour * 24 * 30
+	decisionTaskStartToCloseTimeout     = time.Minute
+	signalName                          = "cadsys-signal-sig"
+	signalsUntilContinueAsNew           = 1000
+	systemWorkflowFnName                = "SystemWorkflow"
+	archivalUploadActivityFnName        = "ArchivalUploadActivity"
+	archivalDeleteHistoryActivityFnName = "ArchivalDeleteHistoryActivity"
+	backfillActivityFnName              = "BackfillActivity"
+	historyBlobKeyExt                   = "history"
+	blobstoreOperationsDefaultTimeout   = 5 * time.Second
+
+	// the following are all non-retryable error strings
+	errArchivalUploadActivityGetDomainStr           = "failed to get domain from domain cache"
```

Comment:
Instead of strings we usually define actual errors which can be used in the code.

Reply:
In this case I define both strings and errors that way I can reference these strings from the activity non-retryable errors config.